### PR TITLE
Fix MutableMap.entries.contains(Map.Entry) 

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**
#KT-42428 Fixed.

### CODE REVIEW

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package test.collections
+
+import kotlin.test.*
+
+// TODO: consider moving to common stdlib tests.
```
[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:10]: The question is: would it be possible to disable this test then for backends that don't support it?

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+class KT42428Test {
+
+    @Test fun testListOfPairsToMapEntriesContainsMapEntry() {
+        val map = ('a'..'z').withIndex().map { (i, c) -> "$c" to i }.toMap()
```
[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:14]: The intention here is to test exactly the case reported in the issue, and I would prefer to keep it that way.
Does it make sense to add additional test case for `associate*` variant?

[the remark this answers is not in the document]
Comment (ilya-g) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:14]: If we have identified the root cause, it doesn't really matter how we create the map.

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilya-g) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:10]: I don't expect if failing on other platforms.

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:14]: > If we have identified the root cause, it doesn't really matter how we create the map

Not sure if I agree here.

> but it's better to be sure that we test that map implementation that caused the problem.

What exactly do you mean?

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:10]: Are you sure? https://youtrack.jetbrains.com/issue/KT-41278

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilya-g) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:14]: The problem was in `HashMap` implementation. It's not guaranteed that `toMap`/`toMutableMap` will always return `HashMap`, so one day this test may stop testing the `HashMap`. It's better to test it both for default implementations of `Map` and `MutableMap`, and for the concrete `HashMap`/`LinkedHashMap`.

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilya-g) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:10]: Oh, yeah. Worth fixing as well.

[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:10]: Well, I might try to fix this later, but not sure it won't require compiler fixes (like https://github.com/JetBrains/kotlin-native/pull/4488) and generally I don't maintain JS stdlib.

Until then, is there a way to include the test to common stdlib tests but disable it for JS IR?
If not, then I propose keeping the test here for now.

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt]: Kinda done.

File: backend.native/tests/stdlib_external/collections/KT42428Test.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilya-g) [reply backend.native/tests/stdlib_external/collections/KT42428Test.kt:10]: I can try fixing it in the same way in JS with an intermediate set class.

**Review by ilya-g — APPROVED** [review]

**Pull request merged**